crates/red_knot_python_semantic/resources/mdtest/annotations/union.md

```diff
@@ -0,0 +1,58 @@
+# Union
+
+## Annotation
+
+`typing.Union` can be used to construct union types same as `|` operator.
+
+```py
+from typing import Union
+
+a: Union[int, str]
+a1: Union[int, bool]
+a2: Union[int, Union[float, str]]
+a3: Union[int, None]
+a4: Union[Union[float, str]]
+
+def f():
+    # revealed: int | str
+    reveal_type(a)
+    # Since bool is a subtype of int we merge both types to int here. But we do allow assigning boolean value
+    # TODO: Pyright and Mypy do show this as `int | bool`: https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmUkQWEMoCqKSYKAUGQIYBchxpA2qjADRQBGYYANgLoUATAKbAowABQBKamShyoIIQDchlbgH14CIeMqSgA
+    # Also mypy: https://mypy-play.net/?mypy=latest&python=3.12&gist=fd3c1609080ddc000d5247552c15b750
```

Comment:
I'm not sure why mypy and pyright do this! I can find plenty of comments in both bug trackers from maintainers clearly stating that `bool` is a subtype of `int`. And even more weirdly, mypy sometimes does simplify `bool | int`: https://mypy-play.net/?mypy=latest&python=3.12&gist=2566490340562ba079a91c53e88c1141

It seems like pyright just generally doesn't simplify subtypes out of unions, it's not specific to `bool | int`: https://pyright-play.net/?code=MYGwhgzhAECCBc0B0KBQpIwEIApYEpEUlVUATAUwDNoqd9oBaAPjmgB9osi1UAnCgDcKYEAH0ALgE8ADhRx18%2BIA

In any case, while interesting, I don't think this is a TODO for us to change, so I would remove this.
```suggestion
```